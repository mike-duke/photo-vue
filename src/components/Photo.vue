
<template>
  <article class="photo" :style="{ backgroundImage: `URL(${photo.url})` }">
    <div class="photo-info">
      <p>{{ photo.description }}</p>
      <ul>
        <li v-for="(tag, index) in photo.tags" :key="index">{{ tag }}</li>
      </ul>
    </div>
  </article>
</template>
<script>

  export default {
    name: 'Photo',
    props: {
      photo: {
        type: Object,
        required: true
      }
    }
  }
</script>
<style>
  .photo {
    height: 400px;
    width: 300px;
    background-size: cover;

  }
</style>