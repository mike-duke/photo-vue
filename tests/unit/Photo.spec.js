import Photo from '@/components/Photo.vue';
import { mount } from '@vue/test-utils';

describe('Photo', () => {
  it('should match the snapshot', () => {
    const wrapper = mount(Photo, {
      propsData: {
        photo: {
          url: 'someplace.com',
          description: 'a picture of stuff',
          photo_tags: [],
          user: {
            name: 'Joe Photo',
            profile_image: 'my ugly mug',
            instagram_username: 'joephoto',
            location: 'Anywhere, USA'
          }
        }
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});