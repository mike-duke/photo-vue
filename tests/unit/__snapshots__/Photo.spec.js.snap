// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should match the snapshot 1`] = `
<article class="photo">
  <div class="user-name"><img src="my ugly mug" alt="Joe Photo" class="user-image">
    <p>Photo by Joe Photo</p>
  </div>
  <div class="image"><img src="someplace.com" alt="a picture of stuff" class="photo-img"></div>
  <div class="photo-info">
    <p>instagram: joephoto</p>
    <p>location: Anywhere, USA</p>
    <p>Tags:
      <ul></ul>
    </p>
  </div>
</article>
`;
